pkgs/applications/misc/openbrf/default.nix

```diff
@@ -12,12 +12,22 @@ mkDerivation {
     sha256 = "16254cnr60ihcn7bki7wl1qm6gkvzb99cn66md1pnb7za8nvzf4j";
   };
 
-  buildInputs = [ qtbase vcg glew ];
+  patches = [
+    # split BoundaryWeight into BoundaryQuadricWeight and QualityQuadricWeight
+    # while preserving the original values where used.
+    # old: https://github.com/cnr-isti-vclab/vcglib/blob/a8e87662b63ee9f4ded5d4699b28d74183040803/vcg/complex/algorithms/local_optimization/tri_edge_collapse_quadric.h#L491
+    # new: https://github.com/cnr-isti-vclab/vcglib/blob/e4950d12e2db7f6ec3e587b06b2a0474b4f08d96/vcg/complex/algorithms/local_optimization/tri_edge_collapse_quadric.h#L590
+    ./vcgmesh.diff
```

Comment:
Could you open an issue on the openbrf upstream and refer to it?

Reply:
an issue or a PR? the repo is dead, last commit is from 2016 with no outstanding PRs and the only issue looks like spam for pokerai in 2019.

I could also just use the older version of the library. eg:

```nix
let
  vcg-1_0_1 = vcg.overrideAttrs(final: prev: {
    version = "1.0.1";
    src = fetchFromGitHub {
      owner = "cnr-isti-vclab";
      repo = "vcglib";
      rev = "v${final.version}";
      hash = "sha256-Tvxe88izIxnIZNpTmgfTUExHTsED4JExiiwfmxGTCEo=";
    };
  });
in
mkDerivation {
```

which doesn't need a patch. perhaps that is better. idk.

Reply:
It's not archived, so they may respond warmly to a PR. Niche tools like these generally don't receive updates unless there are breakages, and package distributions like nixpkgs is where such breakages are commonly discovered.

Skimming the fix, i understand this is an outdated vendored library?

Reply:
> Skimming the fix, i understand this is an outdated vendored library?

yes, vcglib 1.0.1 is from 2016. it was updated in nixpkgs this year to something circa 2022 which resulted in the breakage of the openbrf package.

Reply:
not sure if i understood the question.

Reply:
I was just trying to understand why it seemed vcglib is seemingly copied into the openbrf tree.

Reply:
I decided to just go with using the older version of vcglib. it is a self contained header only package.

Reply:
ok